httpcache.go

```diff
@@ -498,6 +506,44 @@ func headerAllCommaSepValues(headers http.Header, name string) []string {
 	return vals
 }
 
+type cachedReadCloser struct {
```

Comment:
This struct definitely needs to be documented. What is its purpose? What does it do? What are the requirements for using it correctly?